tasks/post-to-cmr/tests/cmr-test.js

```diff
@@ -118,19 +119,89 @@ test.serial('postToCMR returns SIT url when CMR_ENVIRONMENT=="SIT"', async (t) =
   }
 });
 
-test.serial('postToCMR skips CMR step if the metadata file uri is missing', async (t) => {
+test.serial('postToCMR throws an error if there is no CMR metadata file', async (t) => {
   const newPayload = t.context.payload;
 
   newPayload.input.granules = [{
     granuleId: 'some granule',
     files: [{
-      filename: `s3://${t.context.bucket}/to/file.xml`
+      filename: `s3://${t.context.bucket}/to/file.blah`
     }]
   }];
 
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+  }
+});
+
+test.serial('postToCMR throws an error if any granule is missing a metadata file', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = {
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.blah`
+    }]
+  };
+  newPayload.input.granules.push(newGranule);
+
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+    t.is(error.message, (`CMR Meta file not found for granule ${newGranule.granuleId}`));
+  }
+});
+
+test.serial('postToCMR continues without metadata file if there is skipMetaCheck flag', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = [{
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.blah`
+    }]
+  }];
+  newPayload.input.granules = newGranule;
+  newPayload.config.skipMetaCheck = true;
+  const granuleId = newPayload.input.granules[0].granuleId;
+
```

Comment:
Can we wrap `postToCMR` in a try/catch and do `t.fail` in the catch? Just makes it more explicit that we should fail this test if an error is thrown by `postToCmr`

Reply:
This specific test shouldn't fail because we've added the `skipMetaCheck` flag

Reply:
Right, I know the test shouldn't fail. We  expect an error not to be thrown. Right now, if an error was thrown, AVA would auto-fail the test due to an unhandled exception being thrown. But I'm saying it's marginally better to be explicit with a try/catch block and do a `t.fail` in the catch block if an error is thrown, rather than relying on AVA to do teh right thing with unhandled exceptions